bundler/lib/bundler/spec_set.rb

```diff
@@ -202,6 +202,7 @@ def specs_for_dependency(dep, platform)
       specs_for_name = lookup[dep.name]
       target_platform = dep.force_ruby_platform ? Gem::Platform::RUBY : (platform || Bundler.local_platform)
       matching_specs = GemHelpers.select_best_platform_match(specs_for_name, target_platform)
+      matching_specs.each {|s| s.force_ruby_platform = true } if dep.force_ruby_platform
```

Comment:
Hmm, maybe a little refactor could make this solution feel more elegant to you?

```
matching _specs =
  if dep.force_ruby_platform
    GemHelpers.force_ruby_platform(specs_for_name)
  else
    GemHelpers.select_best_platform_match(specs_for_name, platform || Bundler.local_platform)
  end
```

Then you don't need to iterate through matching_specs twice since it can be handled in GemHelpers more cleanly.

Reply:
honestly, feels the same. matching specs should always be a very small array, so the double iteration doesn't feel _that_ bad